### ISSUE TITLE
Features needed before we write publication

### ISSUE THREAD

**Issue by dww100** [UCL-CCS/EasyVVUQ#120]
I'm canvasing opinion here to see if there is anything we have missed.

Things which have been mentioned:

1. Choices/categorical variables
2. Replication sampler
3. Vector parameters
4. Physical checking of parameters

The first two are already included in the [paper plan](https://github.com/UCL-CCS/EasyVVUQ/projects/6). I think it would be worth checking if those above or any other features seem key to people.

**Comment by djgroen** [comment]
@dww100 I think 3 would be nice, but is for me non-essential. As for (4), I think that one would be best reserved for a future occasion, as from my experience with (A)MUSE those kind of things can become quite complicated.

In terms of examples, I notice that there is a Fusion and Climate example. Perhaps it would be nice to include a third example as well, to better showcase the general nature of the toolkit?

**Comment by dww100** [comment]
@djgroen I think the basics of 4 are pretty easy - basically we need to check that defined values are between the input values given by the user. I agree that more elaborate (unit based for example) checks are beyond the scope of this version.

In terms of how essential 3 is - I guess @jlakhlili is the person to say where this should be in our priority list.

There is a third example - a mini-project we have with @velocirobbie and Irina in CCS on box size influence on MD observables. The good thing is that this moves in a V&V direction. We could definitely do more but the bases we want covered are:
- Something with explicit HPC simulations (Materials)
- Complex UQ (Climate)
- Uncertainty propagation (Fusion, hopefully)

**Comment by djgroen** [comment]
Okay, that looks like a sensible set up to me. :)

**Comment by raar1** [comment]
Note that with regards to no. 1, I've already gone some way towards that with the implementation of a sweep sampler (#119) which allows parameter sweeps over any number of category vars. What we don't have, however, is a way to have a parameter sweep within which a stochastic sampler varies the other params. We can get round to that if it's essential for the paper.

**Comment by raar1** [comment]
I have implemented physical range checking using Cerberus (see issue #51), so no. 4 is done.

**Comment by raar1** [comment]
Depending on what exactly we now mean by "vector parameters" (no. 3) this may now already be possible. If it was originally an issue with reading from the JSON file, then that isn't a block anymore since it's fully programmatic at present. And if it's to do with "how do we verify it" then we can already do this with the Cerberus stuff I just implemented.

What remains the sticking point for no. 3?

**Comment by dww100** [comment]
The original issue was I picked up with related to the JSOn input but I did think it would be an issue with verification. If that is fixed then all I would be interested in knowing is how such a parameter should be varied in a sampler and whether we need some sort of block to prevent them getting sampled incorrectly. If we get no reply from @jlakhlili or @wedeling then we should consider this 'supported' for the paper at least.
